src/App.tsx

```diff
@@ -16,46 +17,89 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [isReversed, setReversed] = useState(false);
+  const [sortBy, setSortBy] = useState('');
+  const [goods, setGoods] = useState([...goodsFromServer]);
+
+  const reset = () => {
+    setGoods([...goodsFromServer]);
+    setReversed(false);
+    setSortBy('');
+  };
+
+  const reverse = () => {
+    setGoods([...goods].reverse());
+    setReversed(!isReversed);
+  };
+
+  const handleSort = (goodsList: string[], sortOption: string) => {
+    setSortBy(sortOption);
+    setGoods([...goodsList].sort((g1, g2) => {
+      if (sortOption === 'alphabet') {
+        return g1.localeCompare(g2);
+      }
+
+      return g1.length - g2.length;
+    }));
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames('button', {
+            'is-info is-light': sortBy !== 'alphabet',
+            'is-info': sortBy === 'alphabet',
+          })}
+          onClick={() => {
+            handleSort(goods, 'alphabet');
+          }}
```

Comment:
move to variable. E.g `handleAlphabetSort`
```suggestion
          onClick={handleAlphabetSort}
```